src/patient-search-result/patient-search-result.component.tsx

```diff
@@ -6,72 +6,58 @@ import {
   interpolateString,
   ConfigurableLink,
   ExtensionSlot,
-  useConfig
+  useConfig,
+  fhir
 } from "@openmrs/esm-framework";
 import { SearchedPatient } from "../types";
+import { age } from "../contact-details/age-helpers";
+import { PatientCardProps } from "../patient-card/patient-card-component";
 
 const PatientSearchResults: React.FC<PatientSearchResultsProps> = ({
   patients
 }) => {
   const config = useConfig();
 
-  function renderPatient(patient: SearchedPatient) {
-    const preferredIdentifier =
-      patient.identifiers.find(i => i.preferred) || patient.identifiers[0];
+  function renderPatient(patient: fhir.Patient) {
+    const birthDate = dayjs(patient.birthDate).format("DD - MMM - YYYY");
+
+    const state: PatientCardProps = {
```

Comment:
This state is recreated on render, too. You should make it stable between renders.

Reply:
This is called on a loop and am having issues using memo here.. isn't one of the React hooks rules not to use them on loops? I might be confused.

Reply:
Yes, but you can outsource `renderPatient` to a component `Patient` which takes `patient` as prop (or maybe uses `patient`s properties as props).